www/src/js/utils/timify.ts

```diff
@@ -82,13 +84,21 @@ export const DEFAULT_LATEST_TIME: LessonTime = '1800';
 // This bounds will then be used to decide the starting and ending hours of the timetable.
 export function calculateBorderTimings(
   lessons: Lesson[],
+  period?: TimePeriod,
 ): { startingIndex: number; endingIndex: number } {
   let earliestTime: number = convertTimeToIndex(DEFAULT_EARLIEST_TIME);
   let latestTime: number = convertTimeToIndex(DEFAULT_LATEST_TIME);
   lessons.forEach((lesson) => {
     earliestTime = Math.min(earliestTime, convertTimeToIndex(lesson.StartTime));
     latestTime = Math.max(latestTime, convertTimeToIndex(lesson.EndTime));
   });
+
+  // Consider time range of period, if applicable
+  if (period !== undefined) {
```

Comment:
```suggestion
  if (period != null) {
```
This handles both `undefined` and `null`, which prevents footguns from other developers.